### PR TITLE
Formatter: Treat null and empty array the same

### PR DESCRIPTION
**Pull request by mathias-lang-sociomantic**
This was originally different but it proved to be more of a problem
than an help. Reverting to the old Tango Layout's behavior.

Fixes sociomantic-tsunami/ocean#171

### CODE REVIEW

File: src/ocean/text/convert/Formatter.d [hunk 1]
```diff
@@ -610,6 +604,34 @@ private template DropTypedef (T)
...
+    Before calling `toString`, a `null` check should be performed.
+    However, it shouldn't always be performed, as `T` might not be a ref type.
+
+    Params:
+        v = Object to check, might be a reference type
+        expr = Expression to call if `v` is not null or not a ref type
+        onNull = Expression to call if `v` is null
+
+    Returns:
+        Return type of `expr`.
```
Comment (mihails-strasuns-sociomantic) [inline src/ocean/text/convert/Formatter.d]: I don't think D functions return types ;)

**Review by mihails-strasuns-sociomantic — APPROVED** [review]
Looks sane, would be nice to get confirmation from @don-clugston-sociomantic that it matches old format.

Big question is target branch though. Technically it is a breaking change from current head state. But current state is a bug for certain projects.

**Comment by mathias-lang-sociomantic** [comment]
> Big question is target branch though. Technically it is a breaking change from current head state.

I was wondering the same. Making it target v4.x.x will put it too far in the future. And it's too big of a change to go in a patch release.

File: src/ocean/text/convert/Formatter.d (hunk 1 above)
Comment (mathias-lang-sociomantic) [reply src/ocean/text/convert/Formatter.d]: Corrected

**Comment by don-clugston-sociomantic** [comment]
> Looks sane, would be nice to get confirmation from @don-clugston-sociomantic that it matches old format.

Confirmed with ocean 3.1.7 since it seems be the newest ocean release that all of our other libs support without generating warnings/deprecation messages.
Without this issue fixed, I am stuck in ocean 2.x.

**Comment by mihails-strasuns-sociomantic** [comment]
Ok, I think I will merge it into v2.x.x in that case and we will resort to tagging it as `v2.10.0+breaking` (CC @leandro-lucarella-sociomantic ) to warn about exception as discussed ages ago in some other context :)

**Comment by don-clugston-sociomantic** [comment]
Can we have this merged into a ocean v3 release? For D2 conversion I cannot use anything earlier than ocean v3.1, but this issue prevents upgrading to v3.

It may be a breaking change with regard to ``Format``, but it causes a regression for logging.

**Comment by nemanja-boric-sociomantic** [comment]
This is already released as v3.2.5, @don-clugston-sociomantic .

**Comment by nemanja-boric-sociomantic** [comment]
Ignore my previous comment, I thought I was on the different PR.

**Comment by mihails-strasuns-sociomantic** [comment]
@don-clugston-sociomantic it is already in v3.x.x - https://github.com/sociomantic-tsunami/ocean/commits/v3.x.x , I merged head of v2.x.x into it few days ago. Not in any released version yet though.

**Pull request merged**